objects/src/accounts/package/storage_entry/word.rs

```diff
@@ -0,0 +1,266 @@
+use alloc::{
+    boxed::Box,
+    collections::BTreeMap,
+    string::{String, ToString},
+    vec::Vec,
+};
+use core::fmt;
+
+use serde::{
+    de::{Error as DeError, SeqAccess, Unexpected, Visitor},
+    ser::SerializeSeq,
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use vm_core::{Felt, FieldElement, Word};
+use vm_processor::Digest;
+
+use super::{TemplateKey, TemplateValue};
+use crate::{accounts::package::AccountComponentTemplateError, utils::parse_hex_string_as_word};
+
+// WORDS
+// ================================================================================================
+
+/// Supported word representations in TOML format. Represents slot values and keys.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum WordRepresentation {
+    /// A word represented by a hexadecimal string.
+    SingleHex([Felt; 4]),
+    /// A word represented by its four base elements.
+    Array([FeltRepresentation; 4]),
+    /// A template written as "{{key}}".
+    Dynamic(TemplateKey),
+}
+
+impl WordRepresentation {
+    /// Returns an iterator over all `TemplateKey` references within the `WordRepresentation`.
+    pub fn template_keys(&self) -> Box<dyn Iterator<Item = &TemplateKey> + '_> {
+        match self {
+            WordRepresentation::Array(array) => {
+                Box::new(array.iter().flat_map(|felt| felt.template_keys()))
+            },
+            WordRepresentation::Dynamic(template_key) => Box::new(std::iter::once(template_key)),
+            WordRepresentation::SingleHex(_) => Box::new(std::iter::empty()),
+        }
+    }
+
+    /// Attempts to convert the [WordRepresentation] into a [Word].
+    ///
+    /// If the representation is dynamic, the value is retrieved from `template_values`, identified
+    /// by its key. If any of the inner elements within the value are dynamic, they are retrieved
+    /// in the same way.
+    pub fn try_into_word(
+        &self,
+        template_values: &BTreeMap<String, TemplateValue>,
+    ) -> Result<Word, AccountComponentTemplateError> {
+        match self {
+            WordRepresentation::SingleHex(word) => Ok(*word),
+            WordRepresentation::Array(array) => {
+                let mut result = [Felt::ZERO; 4];
+                for (index, felt_repr) in array.iter().enumerate() {
+                    result[index] = felt_repr.clone().try_into_felt(template_values)?;
+                }
+                Ok(result)
+            },
```

Comment:
Would be good to add a comment here that every index of `result` is guaranteed to be overwritten.

Reply:
Added this